### ISSUE TITLE
Warn about void interpolation expressions

### ISSUE THREAD

**Issue by chalin** [dart-lang/angular_analyzer_plugin#552]
IMHO, an issue should be reported by the analyzer when the static type of an interpolated expression is `void` as in the following example:

```dart
@Component(
  selector: 'example',
  template: '{{foo}}',
)
class C {
  void foo() {}
}
```

cc @kwalrath

**Comment by MichaelRFairhurst** [comment]
Nit: it won't be void unless you bind `{{foo()}}`, but yes, agreed!

**Comment by chalin** [comment]
(Right! I've fixed the example.)
